### PR TITLE
Escape schema name in queue address

### PR DESCRIPTION
**Pull request by kbaley**
Connects to #155 

This ensures the schema name in the queue address is escaped to help distinguish it from other components (e.g. catalog) when we add them in the future.

### CODE REVIEW

File: src/NServiceBus.SqlServer/Addressing/QueueAddress.cs [hunk 1]
```diff
@@ -20,6 +20,10 @@ public static QueueAddress Parse(string address)
                 var parts = address.Split('@');
                 var tableName = parts[0];
                 var schemaName = parts[1];
+                if (schemaName.StartsWith("[") && schemaName.EndsWith("]"))
+                {
+                    schemaName = schemaName.Substring(1, schemaName.Length - 2);
```
Comment (MarcinHoppe) [inline src/NServiceBus.SqlServer/Addressing/QueueAddress.cs]: Could this be handled by regex?

File: src/NServiceBus.SqlServer/Addressing/QueueAddress.cs [hunk 2]
```diff
@@ -29,7 +33,14 @@ public static QueueAddress Parse(string address)
 
         public override string ToString()
         {
-            return $"{TableName}@{SchemaName}";
+            var escapedSchema = SchemaName ?? "";
+            if (!string.IsNullOrWhiteSpace(escapedSchema) && !(escapedSchema.StartsWith("[") && escapedSchema.EndsWith("]")))
```
Comment (tmasternak) [inline src/NServiceBus.SqlServer/Addressing/QueueAddress.cs]: Can we normalize schema in constructor. This way in the whole class we could make assumption that the schema does not contain brackets? We could also extract `UnEscape` method that we could use both in constructor in line 23.

File: src/NServiceBus.SqlServer.UnitTests/QueueAddressTests.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+
+        [Test]
+        [TestCase("table", "schema", "schema")]
+        [TestCase("my.table", "my.schema", "my.schema")]
+        [TestCase("table", "", null)]
+        [TestCase("table", null, null)]
+        [TestCase("table", "[my.schema]", "my.schema")]
+        [TestCase("[my.table]", "[my.schema]", "my.schema")]
+        public void Should_parse_address(string tableName, string schemaName, string expectedSchema)
```
Comment (tmasternak) [inline src/NServiceBus.SqlServer.UnitTests/QueueAddressTests.cs]: I'm wondering if passing whole transport address string and removing `QueueAddress` constructor would not be better here. Right now this test checks that `QueueAddress` is able to parse what it produces but in practice the address will come from the message mappings. 
In other words what I'm saying is let change signature of `Shoudl_parse_address` to (string transportAddress, string expectedSchema).

**Comment by tmasternak** [comment]
I've done my review.

File: src/NServiceBus.SqlServer/Addressing/QueueAddress.cs (hunk 2 above)
Comment (kbaley) [inline src/NServiceBus.SqlServer/Addressing/QueueAddress.cs]: That brings up an interesting point. What if the schema actually _is_ something like [my.schema] with the brackets? As it stands, the `Parse` method would not work.

We don't actually support this now (it creates oddly named queue tables) so my feeling is that we could get away with doing as @tmasternak suggests and normalize in the constructor. I'll make that refactoring

File: src/NServiceBus.SqlServer.UnitTests/QueueAddressTests.cs (hunk 3 above)
Comment (kbaley) [inline src/NServiceBus.SqlServer.UnitTests/QueueAddressTests.cs]: Thanks for the clarification. My intent with this was specifically to test that we could parse what `QueueAddress` produced. Didn't realize that wouldn't happen much in practice. I'll modify this to take in the full transport address.

**Comment by tmasternak** [comment]
@kbaley I've did some small changes. Mainly formatting and naming formatting.

**Comment by kbaley** [comment]
Thanks @tmasternak. Saw your changes and will make note of that for future reference.

**Pull request merged**